src/components/singlEventPageComponents/EventDetails.jsx

```diff
@@ -1,7 +1,150 @@
+                        </div>
+                    </div>
+                    <div className='mt-12 sm:mt-16 lg:mt-0'>
+                        <div className='-ml-48 pr-6 md:-ml-16 lg:relative lg:m-0 lg:h-full lg:px-0'>
+                            <Image
+                                src={eventDisplay.image}
+                                alt='Image 1'
+                                width={647}
+                                height={486}
+                                layout='responsive'
+                                className='w-full rounded-xl shadow-2xl ring-1 ring-black ring-opacity-5 lg:absolute lg:left-0 lg:h-full lg:w-auto lg:max-w-none'
+                            />
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div className='relative'>
+                <div className='lg:mx-auto lg:grid lg:max-w-7xl lg:grid-flow-col-dense lg:grid-cols-2 lg:gap-24 lg:px-8 '>
+                    <div className='mx-auto max-w-xl px-6 lg:mx-0 lg:max-w-none lg:py-16 lg:px-0 '>
+                        <div>
+                            <div className='mt-6'>
+                                <h2 className='text-3xl font-bold tracking-tight text-white'>
+                                    {eventDisplay.title}
+                                </h2>
+                                <p className='mt-4 text-lg text-gray-300'>
+                                    You want to be part of something Bigger !?
+                                </p>
+                            </div>
+                        </div>
+                    </div>
+                    <div className='mt-12 sm:mt-16 lg:mt-0'>
+                        <div className='-mr-48 pl-6 md:-mr-16 lg:relative lg:m-0 lg:h-full lg:px-0'>
+                            <Image
+                                src={eventDisplay.image}
+                                alt='Image 1'
+                                width={647}
+                                height={486}
+                                layout='fixed'
+                                className='w-full rounded-xl shadow-2xl ring-1 ring-black ring-opacity-5 lg:absolute lg:left-0 lg:h-full lg:w-auto lg:max-w-none'
+                            />
+                        </div>
+                        <div className='mt-6'>
+                            <Link
+                                className='inline-flex rounded-lg bg-pink-600 px-4 py-1.5 text-base font-semibold leading-7 text-white shadow-sm ring-1 ring-pink-600 hover:bg-pink-700 hover:ring-pink-700'
+                                href='/'
+                            >
+                                Join the Event !
+                            </Link>
+                        </div>
+                        {/* this one should be only linked to the user who created the event */}
+                        <div className='mt-6'>
+                            <Link
```

Comment:
so you did mention it in the comment line 133
this is the button that should conditioned according to if current.user === event.creator.user
if yes show
else hide

Reply:
i'll make sure to add it

Reply:
i have a solid idea how to do this, but in order us to make work, we a field in the event document that refers to the event creator, like owner, or creator, i'm going/using the same logic for the edit profile in profile page it's something like this
`
 const isOwner = profile && profile.userId === user.uid; `
 
 then in the return statment {isOwner ? (show button) : (hide button)}